webpack.config.js

```diff
@@ -14,18 +25,407 @@ for (var i = 0; i < baseViewPaths.length; i++) {
+                                      // file matching this name
+const CONFIG_GETTEXT_REGEX = /^_\(/;
+const JSTEMPLATES_GETTEXT_REGEX = /{{#_}}(.*?){{\/_}}/g;
+const BASE_STATIC_URL = "/static/";
+const MAPBOX_TOKEN = process.env.MAPBOX_TOKEN;
+
+// TODO: add this to .env
+const CLICKY_ANALYTICS_ID = process.env.CLICKY_ANALYTICS_ID || "";
+
+// TODO: add this to .env
+const MAPQUEST_KEY = process.env.MAPQUEST_KEY || "";
+
+// TODO: add this to .env
+const GOOGLE_ANALYTICS_ID = process.env.GOOGLE_ANALYTICS_ID || "";
+
+// Pull out dataset urls from the .env file. We ignore the keys, as they're no
+// longer needed.
+var datasetSiteUrls = {};
+Object.keys(process.env).forEach(function(key) {
+  if (key.endsWith("_SITE_URL")) {
+    datasetSiteUrls[key] = process.env[key];
+  }
+});
+
+// Logging
+var log = function(msg, time) {
+  if (VERBOSE) {
+    console.log(
+      "(STATIC SITE BUILD) ", 
+      msg, 
+      (
+        (time)
+          ? "in " + time + " ms" 
+          : ""
+      )
+    );
+  }
+};
+
+// Timing
+var time = function(fn) {
+  var start = new Date().getTime();
+  fn();
+  var end = new Date().getTime();
+
+  return end - start;
+};
+
+// Gettext object
+var gt = new Gettext();
+
+
+// (2) Register Handlebars helpers and resolve template inheritances
+// -----------------------------------------------------------------------------
+
+// Helper for serializing config objects and injecting them into the index.html
+// file
+Handlebars.registerHelper("serialize", function(json) {
+  if (!json) return false;
+  return JSON.stringify(json);
+});
+
+// Helper for injecting precompiled templates to the index.html file
+var compiledTemplatesOutputPath;
+Handlebars.registerHelper("precompile_jstemplates", function() {
+  return fs.readFileSync(compiledTemplatesOutputPath);
+});
+
+// Set up template inheritance resolving
+wax.on(Handlebars);
+wax.setLayoutPath(path.resolve(flavorBasePath, "templates"));
+
+
+// (3) Convert the config yaml to json
+// -----------------------------------------------------------------------------
+
+var config;
+var d = time(
+  () => { 
+    config = yaml.safeLoad(fs.readFileSync(flavorConfigPath)); 
+  }
+);
+log("Finished YAML parse", d);
+
+
+// (4) Compile base.hbs and index.html templates for the current flavor
+// -----------------------------------------------------------------------------
+
+var source, template;
+var d = time(
+  () => {
+    source = fs.readFileSync(
+      path.resolve(
+        flavorBasePath, 
+        "templates/index.html"
+      ), 
+      "utf8"
+    );
+    template = Handlebars.compile(source);
+  }
+);
+
+
+// (5) Localize the config for each language for the current flavor, precompile
+//     localized jstemplates Handlebars templates, and inject all localized
+//     content into the index-xx.html file
+// -----------------------------------------------------------------------------
+
+fs.readdirSync(flavorLocaleDir).forEach((langDir) => {
```

Comment:
I think we should move this whole call into a separate script. The handlebars compilation can also be a separate script. It would make this `webpack.config.js` file easier to read and more modular. We'll need to do this anyway if we want to watch file changes and only run specific process of the static site build.